### PR TITLE
TPS-1277: Scrollable table fails to display a dimension column when the first data row is missing that key

### PR DESCRIPTION
**Pull request by mad-raccoon**
# Why is this pull-request needed?

When the clickDimension key was absent from the first row of the result set (e.g. due to a sort order change that moves a sparse row to the top), getTableRows incorrectly stripped that key from every subsequent row that did have it — blanking the entire column.

# Main changes

Changed the guard in getTableRows from checking only rows[0] to rows.some(row => key in row), so the column is preserved as long as any row carries the value.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed an issue in scrollable tables where dimension columns would fail to display when the first data row lacked the corresponding key.

* **Chores**
  * Updated SDK dependencies to latest compatible versions.

* **Tests**
  * Added test coverage for dimension column edge cases.

### CODE REVIEW

**Review by ako-hk — APPROVED** [review]

**Review by yashrajk17 — APPROVED** [review]

File: src/components/charts/tables/tables.utils.tsx [hunk 1]
```diff
@@ -122,7 +122,7 @@ export const getTableRows = (props: { clickDimension?: Dimension; rows: DataResp
 
   const clickDimensionName = props.clickDimension?.name;
 
-  if (!clickDimensionName || Object.keys(props.rows[0]!).includes(clickDimensionName)) {
+  if (!clickDimensionName || props.rows.some((row) => clickDimensionName in row)) {
```
[the remark this answers is not in the document]
Comment (mad-raccoon) [reply src/components/charts/tables/tables.utils.tsx:125]: Not worth it. The rows are JSON data — there's zero chance a column is named toString. I'd just leave the in version as-is and respond to the review comment explaining that the data source makes prototype collision impossible. The comment is a valid static analysis observation but not an actionable bu
